tests/phpunit/tests/blocks/renderCommentTemplate.php

```diff
@@ -137,6 +137,63 @@ function test_build_comment_query_vars_from_block_no_context() {
 		);
 	}
 
+	/**
+	 * Test that if pagination is set to display the last page by default (i.e. newest comments),
+	 * the query is set to look for page 1 (rather than page 0, which would cause an error).
+	 *
+	 * Regression: https://github.com/WordPress/gutenberg/issues/40758.
+	 *
+	 * @ticket 55658
+	 * @covers ::build_comment_query_vars_from_block
+	 */
+	function test_build_comment_query_vars_from_block_pagination_with_no_comments() {
+		$comments_per_page     = get_option( 'comments_per_page' );
+		$default_comments_page = get_option( 'default_comments_page' );
+
+		update_option( 'comments_per_page', 50 );
+		update_option( 'previous_default_page', 'newest' );
+
+		$post_without_comments = self::factory()->post->create_and_get(
+			array(
+				'post_type'    => 'post',
+				'post_status'  => 'publish',
+				'post_name'    => 'fluffycat',
+				'post_title'   => 'Fluffy Cat',
+				'post_content' => 'Fluffy Cat content',
+				'post_excerpt' => 'Fluffy Cat',
+			)
+		);
+
+		$parsed_blocks = parse_blocks(
+			'<!-- wp:comment-template --><!-- wp:comment-author-name /--><!-- wp:comment-content /--><!-- /wp:comment-template -->'
+		);
+
+		$block = new WP_Block(
+			$parsed_blocks[0],
+			array(
+				'postId' => $post_without_comments->ID,
+			)
+		);
+
+		$this->assertSameSetsWithIndex(
+			array(
+				'orderby'       => 'comment_date_gmt',
+				'order'         => 'ASC',
+				'status'        => 'approve',
+				'no_found_rows' => false,
+				'post_id'       => $post_without_comments->ID,
+				'hierarchical'  => 'threaded',
+				'number'        => 50,
+				'paged'         => 1,
+			),
+			build_comment_query_vars_from_block( $block )
+		);
+
+		update_option( 'comments_per_page', $comments_per_page );
+		update_option( 'default_comments_page', $default_comments_page );
```

Comment:
If the test fails, ie `assertSameSetsWithIndex()` fails, this cleanup work will not happen. It should be part of a `tear_down()` process. With that said, doesn't the tear down process already reset the options? Hmm, need to check.

Reply:
Looking at the other tests in the same file, this same cleanup pattern exists. The test in this PR is consistent with the other tests in the test class.

Conclusion: this test is fine to go in as is. Then the test class as a whole can be fixed in a separate PR ... later.

Reply:
Thanks a lot for confirming @hellofromtonya!